corehq/apps/users/bulkupload.py

```diff
@@ -438,8 +438,10 @@ def is_password(password):
                         error = custom_data_validator(data)
                         if error:
                             raise UserUploadError(error)
+                        data = {k: unicode(v) for k, v in data.iteritems()}
                         user.user_data.update(data)
                     if uncategorized_data:
+                        uncategorized_data = {k: unicode(v) for k, v in uncategorized_data.iteritems()}
```

Comment:
I do know some projects store barcodes or fingerprint info in user data, but I believe those will be string-encoded anyways, as I can't imagine we parse user input and store it as binary data.